pkg/apis/gcp/database/v1alpha1/cloudsql_instance_types.go

```diff
@@ -98,6 +103,8 @@ type CloudsqlInstance struct {
 	Status CloudsqlInstanceStatus `json:"status,omitempty"`
 }
 
+var _ resource.Managed = &CloudsqlInstance{}
```

Comment:
Yes, you've mentioned this before. I think the reason why I like it here is mostly due to compile time check validation.

Reply:
I had not considered that - I see that if this test is in a test file it goes unnoticed by `go build` due to the tests not being linked in that scenario.

On the other hand(s):

* This _is_ a test, but it's not very obvious that it is a test. The first time I saw this pattern I had to look it up to figure out what it was doing. I feel that putting it in a test file gives a bit of a clue that it's testing something. This could also be done with a comment.
* Our `make build` will fail even if this is in a test file, because it causes the linter to fail.
* We gate every PR on the tests passing, so I don't think we're really taking much risk by moving it into a test.

This isn't a hill I want to die on, but it would be nice to have some consistency in where we put these. I've been moving them into test files whenever I find them so we may be working at odds.

Reply:
I don't mind moving it to `test`. With that said,

> This is a test, but it's not very obvious that it is a test. The first time I saw this pattern I had to look it up to figure out what it was doing. I feel that putting it in a test file gives a bit of a clue that it's testing something. This could also be done with a comment.

I think I had a similar reaction when I saw it for the first time. However, once I understood the meaning of this, I liked it, and as you can tell: embraced it. I like to assert my "interface-noncompliance" at the compile time.  

> Our make build will fail even if this is in a test file, because it causes the linter to fail.

I avoid `make build` until the very-very-very end (for various reason), i.e. right before pushing my code to remote.

> We gate every PR on the tests passing, so I don't think we're really taking much risk by moving it into a test.

I agree, however, when I am refactoring or authoring `interfaces` or the `implementations`, I think I prefer a quicker feedback loop in the form of `compilation failure` (or even "red squiggly line"), rather than get the same information from the `test` (local or CI/CD).